app/models/package.rb

```diff
@@ -398,6 +401,12 @@ def validate_set_id
     errors.add(:errors, I18n.t('package_sets.no_box_in_set')) if package_set_id.present? && storage_type&.aggregate?
   end
 
+  def validate_set_member
+    if package_set.presence && package_type != package_set.package_type
+      errors.add(:errors, "Invalid package set")
+    end
```

Comment:
this condition `package_type != package_set.package_type` is not enough.

A package can have the same type as its set, OR a children type. That condition doesn't take into account the children types